capa/main.py

```diff
@@ -773,65 +894,157 @@ def get_signatures(sigs_path):
     return paths
 
 
+def get_sample_hashes(sample_path, extractor: FeatureExtractor) -> Tuple[str, str, str]:
+    if isinstance(extractor, StaticFeatureExtractor):
+        md5_ = hashlib.md5()
+        sha1_ = hashlib.sha1()
+        sha256_ = hashlib.sha256()
+
+        with open(sample_path, "rb") as f:
+            buf = f.read()
+
+        md5_.update(buf)
+        sha1_.update(buf)
+        sha256_.update(buf)
+
+        md5, sha1, sha256 = md5_.hexdigest(), sha1_.hexdigest(), sha256_.hexdigest()
+    elif isinstance(extractor, DynamicFeatureExtractor):
+        import json
+
+        if isinstance(extractor, capa.features.extractors.cape.extractor.CapeExtractor):
+            with open(sample_path, "rb") as f:
+                report = json.load(f)
+            md5 = report["target"]["file"]["md5"]
+            sha1 = report["target"]["file"]["sha1"]
+            sha256 = report["target"]["file"]["sha256"]
+        else:
+            md5, sha1, sha256 = "0", "0", "0"
+    else:
+        raise ValueError("invalid extractor")
+
+    return md5, sha1, sha256
+
+
+def get_sample_analysis(format_, arch, os_, extractor, rules_path, counts):
```

Comment:
that sounds like a good idea to me. perhaps a `get_sample_hashes()` method in the `StaticFeatureExtractor` and `DynamicFeatureExtractor` classes?

Reply:
that's what I was thinking

Reply:
I haven't added the `get_sample_analysis()` logic into the extractor classes, since I think it would generate a lot of duplicate code across extractors. I added only the `get_sample_hashes()` logic there since it's pretty dependent on the extractor (different hash info placement in dynamic reports, and different api's for extractor backends).